packages/server/src/controllers/user.ts

```diff
@@ -0,0 +1,170 @@
+//  Required Libraries
+import jwt from 'jsonwebtoken';
+import axios, { AxiosResponse } from 'axios';
+import * as nodemailer from 'nodemailer';
+import * as crypto from 'crypto';
+import { Request, Response } from 'express';
+//  DB Model
+import User from '../models/user.js';
+
+//  Register User
+export const registerUser = async (req: Request, res: Response) => {
+  try {
+    const { email } = req.body;
+    const existingUser = await User.findOne({ email });
+    if (existingUser) { return res.status(400).send('User already Exists!'); }
+    // Generate a random 6-digit OTP
+    const otp = crypto.randomBytes(4).toString('hex');
+    // Save OTP in a new collection. To be implemented
+    const user = new User({
+      email,
+      password: null,
+      files: [],
+      otp,
+      isVerified: false,
+      secret: null,
+    });
+    await user.save();
+
+    const dataToken = jwt.sign({ userID: email, isValid: false, role: 'user' }, process.env.JWT_SECRET as string);
+
+    // Send the OTP to the user's email
+    const transport = nodemailer.createTransport({
```

Comment:
Move this `transport` function out of the `registerUser` function. As right now this transport function is being created again and again every time this function runs.
eg:
```javascript
const transport = ...

const registerUser = (...) => {
  ...
  transport.sendMail(...)
}
```